includes/datastore.inc.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/*
+ * LibreNMS abstract data storage interface to both rrdtool & influxdb
+ *
+ * Copyright (c) 2016 Paul D. Gear <paul@librenms.org>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+require_once $config['install_dir'] . "/includes/rrdtool.inc.php";
+require_once $config['install_dir'] . "/includes/influxdb.inc.php";
+
+
+/*
+ * @return Copy of $arr with all keys beginning with 'rrd_' removed.
+ */
+function rrd_array_filter($arr)
```

Comment:
You can pass array_filter a mode that only takes care of keys rather than values. see:
https://secure.php.net/manual/en/function.array-filter.php
```php
$new = array_filter($array,function() {...}, ARRAY_FILTER_USE_KEY);
```